Source/SwiftLintBuiltInRules/Rules/Style/MultilineLiteralBracketsRule.swift

```diff
@@ -97,43 +98,79 @@ struct MultilineLiteralBracketsRule: ASTRule, OptInRule {
                      4, 5, 6,
                      7, 8, 9
                 ]
+            """),
+            Example("""
+            class Hogwarts {
+                let houseCup = [
+                    "gryffindor": 460, "hufflepuff": 370,
+                    "ravenclaw": 410, "slytherin": slytherinPoints.filter {
+                        $0.isValid
+                    }.sum()↓]
+            }
             """)
         ]
     )
+}
 
-    func validate(file: SwiftLintFile,
-                  kind: SwiftExpressionKind,
-                  dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        guard
-            [.array, .dictionary].contains(kind),
-            let bodyByteRange = dictionary.bodyByteRange,
-            let body = file.stringView.substringWithByteRange(bodyByteRange)
-        else {
-            return []
+private extension MultilineLiteralBracketsRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visitPost(_ node: ArrayExprSyntax) {
+            validate(
+                node,
+                openingToken: node.leftSquare,
+                closingToken: node.rightSquare,
+                firstElement: node.elements.first?.expression,
+                lastElement: node.elements.last?.expression
+            )
         }
 
-        let isMultiline = body.contains("\n")
-        guard isMultiline else {
-            return []
+        override func visitPost(_ node: DictionaryExprSyntax) {
+            switch node.content {
+            case .colon:
+                break
+            case .elements(let elements):
+                validate(
+                    node,
+                    openingToken: node.leftSquare,
+                    closingToken: node.rightSquare,
+                    firstElement: elements.first?.key,
+                    lastElement: elements.last?.value
+                )
+            }
         }
 
-        let expectedBodyBeginRegex = regex("\\A[ \\t]*\\n")
-        let expectedBodyEndRegex = regex("\\n[ \\t]*\\z")
+        private func validate(_ node: any ExprSyntaxProtocol,
```

Comment:
Can this be `some` as well?

```suggestion
        private func validate(_ node: some ExprSyntaxProtocol,
```